mlrun/datastore/sources.py

```diff
@@ -826,6 +829,139 @@ def add_nuclio_trigger(self, function):
+
+    example::
+         connection_string = "???"
+         query = {age: {"$gt: 5}}
+         MongoDBSource(connection_string=connection_string, collection_name="coll",
+         db_name="my_dataset", chunksize=5, query=query)
+
+    :parameter name:  source name
+    :parameter query: dictionary query for mongodb
+    :parameter chunksize: number of rows per chunk (default large single chunk)
+    :parameter key_field: the column to be used as the key for events. Can be a list of keys.
+    :parameter time_field: the column to be parsed as the timestamp for events. Defaults to None
+    :parameter start_time: filters out data before this time
+    :parameter end_time: filters out data after this time
+    :parameter schedule: string to configure scheduling of the ingestion job. For example '*/30 * * * *' will
+         cause the job to run every 30 minutes
+    :parameter db_name: the name of the database to access
+    :parameter connection_string: your mongodb connection string
+    :parameter collection_name: the name of the collection to access,
+                                    from the current database
+    :parameter spark_options: additional spark read options
+    """
+
+    kind = "mongodb"
+    support_storey = True
+    support_spark = False
+
+    def __init__(
+        self,
+        name: str = "",
+        max_results_for_table: int = None,
+        query: dict = None,
+        chunksize: int = None,
+        key_field: str = None,
+        time_field: str = None,
+        schedule: str = None,
+        start_time=None,
+        end_time=None,
+        db_name: str = None,
+        connection_string: str = None,
+        collection_name: str = None,
+        spark_options: dict = None,
+    ):
+        key = "MONGO_CONNECTION_STRING"
+        connection_string = connection_string or os.getenv(key)
+        if connection_string is None:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"cannot specify without connection_string arg or secret {key}"
+            )
+        else:
+            mongodb_client = MongoClient(connection_string)
+        attrs = {
+            "query": query,
+            "collection_name": collection_name,
+            "db_name": db_name,
+            "max_results": max_results_for_table,
+            "chunksize": chunksize,
+            "mongodb_client": mongodb_client,
+            "spark_options": spark_options,
+            "connection_string": connection_string,
+        }
+        attrs = {key: value for key, value in attrs.items() if value is not None}
+        super().__init__(
+            name,
+            attributes=attrs,
+            key_field=key_field,
+            time_field=time_field,
+            schedule=schedule,
+            start_time=start_time,
+            end_time=end_time,
+        )
+
+    def to_dataframe(self):
+        class MongoDBiter:
+            def __init__(self, collection, iter_chunksize, iter_query):
```

Comment:
please add type hints and doc strings to all the methods and variables you added.